crates/core/src/error/collector.rs

```diff
@@ -0,0 +1,289 @@
+use derive_more::From;
+
+use crate::{error::RusticErrorKind, RusticError};
+use log::{error, info, warn};
+
+/// A rustic issue result
+///
+/// rustic issue results are used to return a result along with possible issues.
+#[derive(Debug)]
+pub enum RusticIssueResult<T> {
+    /// For when we return a result along with possible issues
+    Ok(T, Option<Vec<RusticIssue>>),
+
+    /// For when we abort and return only the errors that occurred
+    Err(Vec<RusticError>),
+}
+
+/// A rustic issue
+///
+/// An issue is a message that can be logged to the user.
+#[derive(Debug)]
+pub enum RusticIssue {
+    /// An error issue, indicating that something went wrong irrecoverably
+    Error(RusticError),
```

Comment:
We might not have that within this Enum though? 🤔  as that should always be a hard error I think? Not sure though.

Reply:
Actually I do think that we can have Errors on a deeper level which are shown as warnings on a higher level. Something like backup failed for creating the first snapshot, but we want to continue creating the second one...

Reply:
In this case, we should handle that at the point we get that `error` and start the backup for the second `path` for example. The failed backup for the first will be output to the user at the `match` of the result (e.g. in a loop) and we start the next one, but give user feedback.

Reply:
So I think this is the same topic as above, we shouldn't give `errors` different meanings, I don't know if they are really contextual in that sense. There can be hard and soft errors, for sure. But I think we are going to replace the term soft error here, with an `issue` which can result in a `warning` or `info`. But then we need to reserve the term `error` for things that are irrecoverable.